examples/glfw/run.sh

```diff
@@ -1,14 +1,20 @@
 #!/bin/bash
 set -e # Exit if any program returns an error.
 
+if arch | grep "arm64"; then
```

Comment:
I think you want `uname -m`.  On my Intel machine correctly reports `x86_64`:
```
$ uname -m
x86_64
$ arch
i386
```

On an M1 machine last I checked it output `aarch64, arm64_v8a` but I'm not on one atm to check.

https://github.com/flutter/engine/blob/c6e55648956faf6e7317085fad67efca47c0edab/lib/web_ui/dev/felt#L28
https://github.com/flutter/buildroot/blob/33358ba597f875f6646522fd4ea86c1b31404d77/build/install-chroot.sh#L187
https://github.com/flutter/flutter/blob/fc015f52c6e294af1d27f5aa068cced8c05017c1/packages/flutter_tools/lib/src/base/os.dart#L268-L270

Reply:
Is there a scenario where you don't think this works?  `arch` gives you the settable value for the architecture of your terminal , whereas uname will give the actual cpu.  It seems like respecting the user's setting is the right way instead of hardcoding it.

Reply:
`uname` respects if you are running the session in Rosetta and will use `x86_64` even if you are on an arm machine.  Also if we ever added an else `x86_64` or something it unexpectedly wouldn't work--I definitely never set my terminal arch to `i386`.

Reply:
From Triage: Just checking in to see if we are proceeding on this. From perusing the conversation, we seem to not be sure of what to do when the terminal arch is set to use emulation. Is that correct? Does it matter which arch it is as long as the example runs?

Reply:
I think it should be `uname` as it is in other scripts because of Rosetta, but this works so if @gaaclarke has a strong preference then I don't care if he leaves as-is.

Reply:
I preferred `arch` because:
1) Its name is more apropos
1) It is the same function used for setting target architecture
1) I didn't know if uname respects `arch` settings (but i verified it does)
1) The alternative wasn't convincing that it merited further investment when the status quo worked

It's not a huge deal, jenn has a strong preference so I switched it to `uname`, @jmagman ptal.

Reply:
> It's not a huge deal, jenn has a strong preference so I switched it to `uname`, @jmagman ptal.

To be fair I said I _don't_ have a strong preference 😉 

> but this works so if @gaaclarke has a strong preference then I don't care if he leaves as-is.